File: android/src/main/kotlin/br/dig/dev/flutter/flutter_reference/FlutterReferencePlugin.kt
package br.dig.dev.flutter.flutter_reference

import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result

class FlutterReferencePlugin : FlutterPlugin, MethodCallHandler {
    private val refs = References
    private lateinit var channel: MethodChannel

    override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
        channel = MethodChannel(flutterPluginBinding.binaryMessenger, "flutter_reference")
        channel.setMethodCallHandler(this)
    }

    override fun onMethodCall(call: MethodCall, result: Result) {
        return when (call.method) {
            "dispose" -> handle(call, result) {
                refs.remove(it.arguments<String>()!!)
            }

            else -> result.notImplemented()
        }
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        channel.setMethodCallHandler(null)
        refs.clear()
    }
}

File: ios/Classes/FlutterReferencePlugin.swift
import Flutter
import UIKit

public class FlutterReferencePlugin: NSObject, FlutterPlugin {
  private let refs = References.shared
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(
      name: "flutter_reference", binaryMessenger: registrar.messenger())
    let instance = FlutterReferencePlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "dispose":
      handle(call, result, action: { it in return try refs.remove(it.argument().unwrap()) })
    default:
      result(FlutterMethodNotImplemented)
    }
  }
}
